deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -939,8 +939,8 @@ def gradient_reduction_w_predivide(self, tensor):
 
     def average_tensor(self, tensor):
         if self.overlap_comm:
-            torch.cuda.synchronize()
             stream = self.reduction_stream
+            stream.wait_stream(torch.cuda.default_stream())
```

Comment:
Can you please explain why it waits for `default_stream()`, rather than `current_stream()` before line 942? What do you expect `default_stream()` to be?

Reply:
I think in this context, `current_stream()` is `default_stream()`. I do agree `current_stream()` is a better choice.

Reply:
Sounds good. Sorry for the delayed response.